common/src/main/java/com/wynntils/functions/HorseFunctions.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.functions;
+
+import com.wynntils.core.functions.ActiveFunction;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.utils.InventoryUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.custom.item.ItemStackTransformModel;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import java.util.List;
+import net.minecraft.world.item.ItemStack;
+
+public class HorseFunctions {
+
+    private static WynnItemStack getHorse() {
+        int horseSlot = InventoryUtils.findHorseSlotNum();
+        if (horseSlot == -1) return null;
+        ItemStack is = McUtils.player().getInventory().getItem(horseSlot);
+        if (!(is instanceof WynnItemStack horse)) return null;
+        return horse;
+    }
+
+    public static class HorseLevelFunction extends ActiveFunction<String> {
```

Comment:
Is this really what we want to call "active"? This rather fits the DependantFunction concept (or whatever it is called now)

Reply:
(This applies to every horse function.

Reply:
What I mean is I don't see any "caching", just "active" fetching.

Reply:
So, yes, it is a DependantFunction, not strictly an "active" one. But as we talked about in the previous PR, this does not really make sense. We need to load the manager, before it can start giving real values. So there is really only two kinds of Functions, those that can return a value at any time, with no prereq's, (like `clock`), and those which has some kind of dependency. 

If they need to call into a Manager/Model, then this needs to be loaded.

Reply:
The alternative is to always load the ItemStackTransformerModel. But if we want to be able to *not* have it always loaded, we somehow need to say that it needs loading for this function. And you can't really load and unload managers willy-nilly each time you want to read a value from a function...

Reply:
I'm not sure I follow you completely here, but are you worried about performance issues? In the old code, info boxes were updated every 4th tick. I doubt it will make much of an impact to query the inventory for horse data 5 times a second. If we should start caching this, we'll run into the old:

There's only two hard things in computer science:
* Naming things, and
* Cache invalidation

Reply:
Am I not up to date on ActiveFunction structure? Does it not have a "is cached" field of some sort? We don't seem to update that

Reply:
So in a way I think what you're saying is that there is a conceptual difference between functions that has a `@SubscribeEvent` and gets updated whenever that event happens, and in between that returns the cached value, and a function like this, that goes and fetches the value "fresh" each time it is called, but does so from a Model that needs to be enabled first. I agree with that. But it's just an "implementation detail". 

From the PoV of the code calling the function, both of them has the property that they need to be "enabled" to give valid data back, even if the reason for this differs. (That was why I called them EnableableFunctions originally :))

Reply:
Okay. I think I get it now, but I don't want to reason on something I'm not yet fully familiar with. I will get back to this.

Reply:
I'm open for improvements on the Function design as well. It is not fully clear to me how to solve this in the best way. I made an attempt that I think is okay, but it does not shine like the only obvious way...